PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -83,8 +84,26 @@ function transformReports(inputReportFiles: string[], index: number)
         sarifReports.push(report);
         processResults(inputReportFiles, index);
     } else if (report.toLocaleLowerCase().endsWith(XML_EXTENSION)) {
+        ruleAnalyzerPairs.clear();
+
         const saxStream = sax.createStream(true, {});
         saxStream.on("opentag", function (node) {
+            if (node.name == 'Rule' && !bLegacyReport) {
+                ruleAnalyzerPairs.set(node.attributes.id, node.attributes.analyzer);
+            }
+
+            if (bLegacyReport) {
+                if (node.name == 'DupViol') {
+                    mapToAnalyzer(node, "com.parasoft.xtest.cpp.analyzer.static.dupcode", false);
+                } else if (node.name == 'FlowViol') {
+                    mapToAnalyzer(node, "com.parasoft.xtest.cpp.analyzer.static.flow", false);
+                } else if (node.name == 'MetViol') {
+                    mapToAnalyzer(node, "com.parasoft.xtest.cpp.analyzer.static.metrics", false);
+                } else if (node.name.endsWith('Viol')) {
```

Comment:
Is there other consideration to use endsWith() method? Since there is another violation node - <FuncViol> node, but we don't need to parse it. Here we only care about <StdViol> node.